src/Missings.jl

```diff
@@ -406,7 +405,13 @@ Base.mapreduce_impl(f, op, A::SkipMissingsofArrays, ifirst::Integer, ilast::Inte
         v = op(f(a1), f(a2))
         @simd for i = i:ilast
             @inbounds ai = A[i]
-            ai === missing || @inbounds _anymissingindex(itr.others, i) || (v = op(v, f(ai)))
+            if ai === missing 
```

Comment:
This looks really weird. Was there a problem with the previous version using `||`? (Sorry if my comment about the other block prompted you to change this one too. I think the fact that the other has `return` makes the situation different.)

Reply:
sorry, no problem. Using the `||` for this line, then.

Reply:
Actually, let me confirm that you see we are assigning a variable `v` after the last `||`. 

It feels weird to assign a variable at the end of all these short-circuited booleans.

Reply:
Yeah it's a bit complex. Normally I would have written this as:
```julia
if ai !== missing && @inbounds !_anymissingindex(itr.others, i)
    v = op(v, f(ai))
end
```
But if the compiler doesn't like it, I'd rather use `||` than empty `if` branches.

Reply:
nevermind (sorry for spamming you with this). There is a very slight performance improvement to using the `||` version so I am keeping it.